perfkitbenchmarker/linux_benchmarks/cloud_spanner_ycsb_benchmark.py

```diff
@@ -0,0 +1,185 @@
+cloud_spanner_ycsb:
+  description: >
+      Run YCSB against Google Cloud Spanner.
+      Configure the number of VMs via --num-vms.
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+      vm_count: 1
+  flags:
+    gcloud_scopes: >
+      https://www.googleapis.com/auth/spanner.admin
+      https://www.googleapis.com/auth/spanner.data"""
+
+# As of May 2017, cloud spanner is not included in the latest YCSB release
+# 0.12.0. You can build your own release binary and feed it in with
+# --cloud_spanner_custom_ycsb_release=<url>. You are welcome to use a pre-build
+# binary located at
+# https://storage.googleapis.com/cloud-spanner-ycsb-custom-release/
+# ycsb-cloudspanner-binding-0.13.0.tar.gz
+YCSB_BINDING_TAR_URL = ('https://github.com/brianfrankcooper/YCSB/releases'
+                        '/download/0.13.0/'
+                        'ycsb-cloudspanner-binding-0.13.0.tar.gz')
+REQUIRED_SCOPES = (
+    'https://www.googleapis.com/auth/spanner.admin',
+    'https://www.googleapis.com/auth/spanner.data')
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('table',
+                    'usertable',
+                    'The table name used in the YCSB benchmark.')
+flags.DEFINE_integer('zeropadding',
+                     12,
+                     'The zero padding used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_host',
+                    None,
+                    'The Cloud Spanner host used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_instance',
+                    'ycsb-' + getpass.getuser(),
+                    'The Cloud Spanner instance used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_database',
+                    'ycsb',
+                    'The Cloud Spanner database used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_readmode',
+                    'query',
+                    'The Cloud Spanner read mode used in the YCSB benchmark.')
+flags.DEFINE_integer('cloud_spanner_boundedstaleness',
+                     0,
+                     'The Cloud Spanner bounded staleness used in the YCSB '
+                     'benchmark.')
+flags.DEFINE_integer('cloud_spanner_batchinserts',
+                     1,
+                     'The Cloud Spanner batch inserts used in the YCSB '
+                     'benchmark.')
+flags.DEFINE_string('cloud_spanner_custom_ycsb_release',
+                    None,
+                    'If provided, the URL of a custom YCSB release')
+flags.DEFINE_string('cloud_spanner_desp',
+                    'YCSB',
+                    'The description of the Cloud Spanner instance.')
+flags.DEFINE_integer('cloud_spanner_nodes',
+                     1,
+                     'The number of nodes for the Cloud Spanner instance.')
+flags.DEFINE_string('cloud_spanner_config',
+                    'regional-us-central1',
+                    'The config for the Cloud Spanner instance.')
+flags.DEFINE_string('cloud_spanner_ddl',
```

Comment:
There is a data folder which might be the more appropriate location for the ddl. Please see example https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/data/cassandra/create-ycsb-table.cql.j2